journal/src/main/java/io/camunda/zeebe/journal/file/SegmentFileManager.java

```diff
@@ -0,0 +1,184 @@
+  }
+
+  private MappedByteBuffer mapSegment(
+      final File segmentFile,
+      final JournalSegmentDescriptor descriptor,
+      final Set<OpenOption> extraOptions)
+      throws IOException {
+    final var options = new HashSet<>(extraOptions);
+    options.add(StandardOpenOption.READ);
+    options.add(StandardOpenOption.WRITE);
+
+    try (final var channel = FileChannel.open(segmentFile.toPath(), options)) {
+      final var mappedSegment = channel.map(MapMode.READ_WRITE, 0, descriptor.maxSegmentSize());
+      mappedSegment.order(ENDIANNESS);
+
+      return mappedSegment;
+    }
+  }
+
+  private JournalSegmentDescriptor readDescriptor(final File file) {
+    try (final FileChannel channel = FileChannel.open(file.toPath(), StandardOpenOption.READ)) {
+      final byte version = readVersion(channel, file.getName());
+      final int length = JournalSegmentDescriptor.getEncodingLengthForVersion(version);
+      if (file.length() < length) {
+        throw new CorruptedLogException(
+            String.format(
+                "Expected segment '%s' with version %d to be at least %d bytes long but it only has %d.",
+                file.getName(), version, length, file.length()));
+      }
+
+      final ByteBuffer buffer = ByteBuffer.allocate(length);
+      final int readBytes = channel.read(buffer, 0);
+
+      if (readBytes != -1 && readBytes < length) {
+        throw new JournalException(
+            String.format(
+                "Expected to read %d bytes of segment '%s' with %d version but only read %d bytes.",
+                length, file.getName(), version, readBytes));
+      }
+
+      buffer.flip();
+      return new JournalSegmentDescriptor(buffer);
+    } catch (final IndexOutOfBoundsException e) {
+      throw new JournalException(
+          String.format(
+              "Expected to read descriptor of segment '%s', but nothing was read.", file.getName()),
+          e);
+    } catch (final UnknownVersionException e) {
+      throw new CorruptedLogException(
+          String.format("Couldn't read or recognize version of segment '%s'.", file.getName()), e);
+    } catch (final IOException e) {
+      throw new JournalException(e);
+    }
+  }
+
+  private byte readVersion(final FileChannel channel, final String fileName) throws IOException {
+    final ByteBuffer buffer = ByteBuffer.allocate(1);
+    final int readBytes = channel.read(buffer);
+
+    if (readBytes == 0) {
+      throw new JournalException(
+          String.format(
+              "Expected to read the version byte from segment '%s' but nothing was read.",
+              fileName));
+    } else if (readBytes == -1) {
+      throw new CorruptedLogException(
+          String.format(
+              "Expected to read the version byte from segment '%s' but got EOF instead.",
+              fileName));
+    }
+
+    return buffer.get(0);
+  }
+
+  JournalSegment createSegment(final JournalSegmentDescriptor descriptor, final File segmentFile) {
```

Comment:
:upside_down_face: I would group the "API" of this class together (i.e. non private functions), since private things are implementation details, so someone looking into this class only has to check the top of the file.